src/main/java/sysc4806/graduateAdmissions/controller/RoleController.java

```diff
@@ -0,0 +1,132 @@
+package sysc4806.graduateAdmissions.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import sysc4806.graduateAdmissions.model.Privilege;
+import sysc4806.graduateAdmissions.model.Role;
+import sysc4806.graduateAdmissions.repositories.RoleRepository;
+
+import java.util.Optional;
+
+/**
+ * Controller for CRUD operations on Role
+ *
+ * @author Eric
+ */
+@RestController
+@RequestMapping("/role/")
+public class RoleController {
+    @Autowired
+    RoleRepository repository;
+
+    /**
+     * Get Role by id or return all roles
+     *
+     * @param id optional to find specific role
+     * @return JSON containing Roles
+     */
+    @ResponseBody
+    @GetMapping("")
+    public ResponseEntity getRole(@RequestParam(required=false) Long id) {
+        if(id == null)
+            return ResponseEntity.status(HttpStatus.OK).body(repository.findAll());
+        else
+            return ResponseEntity.status(HttpStatus.OK).body(repository.findById(id));
+    }
+
+    /**
+     * get Role by name
+     *
+     * @param name the name of the Role to be retrieved
+     * @return JSON containing the Role
+     */
+    @ResponseBody
+    @GetMapping("name")
+    public ResponseEntity getRoleByName(@RequestParam String name){
+        return ResponseEntity.status(HttpStatus.OK).body(repository.findByRoleName(name));
+    }
+
+    /**
+     * create a new Role
+     *
+     * @param role the Role to be added to the system
+     * @return ResponseEntity describing the outcome of the operation
+     */
+    @ResponseBody
+    @PostMapping("create")
+    public ResponseEntity createRoleForm(@RequestBody() Role role) {
```

Comment:
nothing about this suggests that there is a from involved, besides the name of the method

Reply:
missed this when changing controller to return JSON, will fix